spec/tree_spec.rb

```diff
@@ -1,14 +1,66 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe Tree do
+  let(:tree) {Tree.new}
+
   it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+    expect(described_class.is_a? Class).to be true
+  end
+
+  it 'on initialize' do
+    expect(tree.age).to eq(0)
+    expect(tree.height).to eq(0.5)
+    expect(tree.any_apples?).to be false
+  end
+
+  it 'on age!' do
+    tree.age!
+    expect(tree.age).to eq(1)
+  end
+
+  it 'on add_apples and any_apples?' do
+    3.times do
+      tree.age!
+    end
+    tree.add_apples
+    expect(tree.any_apples?).to be true
+  end
+
+  it 'on pick_an_apple!' do
+    3.times do
+      tree.age!
+    end
+    while tree.any_apples? do
+      tree.pick_an_apple!
+    end
+    expect(tree.any_apples?).to be false
+  end
+
+  it 'on dead?' do
+    expect(tree.dead?).to be false
```

Comment:
Usually. But I'd say not here.

Here you are unit testing the `dead?` method, so I think it is valuable to explicitly call that method.

I just scanned the project I'm working on and found this other random example. The application sends out an XML message and I need to verify that it encodes an "approval":
```ruby
expect(NCPDP::Response.parse xml).to be_approved
```
It could have been written as like so:
```ruby
expect(NCPDP::Response.parse(xml).approved?).to eq(true)
```